packages/payment-processor/src/payment/erc20.ts

```diff
@@ -159,20 +174,23 @@ export function prepareApproveErc20(
  * the request in its payment currency. Can be used with a Multisig contract.
  * @param request the request to pay
  * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ * @param isEscrow wether to approve the payment proxy or escrow contract
  */
 export function encodeApproveErc20(
   request: ClientTypes.IRequestData,
   signerOrProvider: providers.Provider | Signer = getProvider(),
+  isEscrow = false,
```

Comment:
Same here, not comfortable with that choice. I think escrow is, for now, a minor option, and later probably a separate RN, so I think it's not worth it re-using the same method.